hash_practice/exercises.py

```diff
@@ -2,18 +2,43 @@
 def grouped_anagrams(strings):
     """ This method will return an array of arrays.
         Each subarray will have strings which are anagrams of each other
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(n)
+        Space Complexity: O(n)
     """
-    pass
+    hash_table = {}
+    for str in strings:
+        key = ''.join(sorted(str))
+        if key in hash_table.keys():
+            hash_table[key].append(str)
+        else:
+            hash_table[key] = []
+            hash_table[key].append(str)
+
+    result = []
+    for key, value in hash_table.items():
+        result.append(value)
+    return result
+
 
 def top_k_frequent_elements(nums, k):
     """ This method will return the k most common elements
         In the case of a tie it will select the first occuring element.
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(n)

```

Comment:
Don't forget to account for the call to `sorted` on line 37, which will take O(n * log(n)) time and thus dominate the O(n) parts of the algorithm. Thus, the correct time complexity here is O(n * log(n)).